File: test-projects/backend/06-spring-boot-java/src/main/java/com/example/Application.java
package com.example;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@SpringBootApplication
@RestController
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @GetMapping("/")
    public Map<String, String> home() {
        return Map.of("message", "Spring Boot Java API - Testing Auto-Docker Extension");
    }

    @GetMapping("/health")
    public Map<String, String> health() {
        return Map.of("status", "healthy");
    }
}

File: test-projects/backend/12-kotlin-ktor/src/main/kotlin/com/example/Application.kt
package com.example

import io.ktor.server.engine.*
import io.ktor.server.netty.*
import io.ktor.server.application.*
import io.ktor.server.response.*
import io.ktor.server.routing.*

fun main() {
    embeddedServer(Netty, port = 8080, host = "0.0.0.0") {
        routing {
            get("/") {
                call.respond(mapOf("message" to "Kotlin Ktor API - Testing Auto-Docker Extension"))
            }
            get("/health") {
                call.respond(mapOf("status" to "healthy"))
            }
        }
    }.start(wait = true)
}
